### PR TITLE
remotecfg: introduce the rest of the httpclientconfig block attributes

### PR DESCRIPTION
**Pull request by tpaschalis**
#### PR Description
This PR adds documentation in the remotecfg page for the rest of the squashed fields of the HTTPClientConfig block.
#### Which issue(s) this PR fixes
No issue filed

#### PR Checklist

- [ ] CHANGELOG.md updated (N/A)
- [X] Documentation added
- [ ] Tests updated (N/A)

### CODE REVIEW

File: docs/sources/reference/config-blocks/remotecfg.md [hunk 1]
```diff
@@ -43,6 +43,23 @@ Name             | Type                 | Description
...
+`enable_http2`           | `bool`              | Whether HTTP2 is supported for requests.                      | `true`  | no
+`follow_redirects`       | `bool`              | Whether redirects returned by the server should be followed.  | `true`  | no
+`proxy_url`              | `string`            | HTTP proxy to send requests through.                          |         | no
+`no_proxy`               | `string`            | Comma-separated list of IP addresses, CIDR notations, and domain names to exclude from proxying. | | no
+`proxy_from_environment` | `bool`              | Use the proxy URL indicated by environment variables.         | `false` | no
+`proxy_connect_header`   | `map(list(secret))` | Specifies headers to send to proxies during CONNECT requests. |         | no
+
+ At most, one of the following can be provided:
+ - [`bearer_token` argument](#endpoint-block).
+ - [`bearer_token_file` argument](#endpoint-block).
```
Comment (spartan0x117) [inline docs/sources/reference/config-blocks/remotecfg.md:57]: Are these the right links? I don't see an `endpoint` block in this doc

File: docs/sources/reference/config-blocks/remotecfg.md [hunk 2]
```diff
@@ -43,6 +43,23 @@ Name             | Type                 | Description
...
 `name`           | `string`             | A human-readable name for the collector.          | `""`        | no
+`bearer_token_file`      | `string`            | File containing a bearer token to authenticate with.          |         | no
+`bearer_token`           | `secret`            | Bearer token to authenticate with.                            |         | no
+`enable_http2`           | `bool`              | Whether HTTP2 is supported for requests.                      | `true`  | no
+`follow_redirects`       | `bool`              | Whether redirects returned by the server should be followed.  | `true`  | no
+`proxy_url`              | `string`            | HTTP proxy to send requests through.                          |         | no
+`no_proxy`               | `string`            | Comma-separated list of IP addresses, CIDR notations, and domain names to exclude from proxying. | | no
+`proxy_from_environment` | `bool`              | Use the proxy URL indicated by environment variables.         | `false` | no
+`proxy_connect_header`   | `map(list(secret))` | Specifies headers to send to proxies during CONNECT requests. |         | no
+
```
Comment (spartan0x117) [inline docs/sources/reference/config-blocks/remotecfg.md:54]: Should we add the proxy config description?
```suggestion

{{< docs/shared lookup="reference/components/http-client-proxy-config-description.md" source="alloy" version="<ALLOY_VERSION>" >}}

```

Comment (tpaschalis) [reply docs/sources/reference/config-blocks/remotecfg.md:54]: Good call, thanks.

File: docs/sources/reference/config-blocks/remotecfg.md (hunk 1 above)
Comment (tpaschalis) [reply docs/sources/reference/config-blocks/remotecfg.md]: Whoops, fixed. I now have them point to the arguments table.

File: docs/sources/reference/config-blocks/remotecfg.md (hunk 2 above)
Comment (tpaschalis) [reply docs/sources/reference/config-blocks/remotecfg.md:54]: I've moved them to the bottom, after the (I imagine) more commonly-used fields.

**Review by spartan0x117 — APPROVED** [review]
LGTM!

**Pull request merged**